kratos/python_scripts/assign_vector_variable_process.py

```diff
@@ -57,41 +48,18 @@ def __init__(self, Model, settings ):
 
         import assign_scalar_variable_process
 
-        #component X
-        if(not settings["value"][0].IsNull()):
-            x_params = KratosMultiphysics.Parameters("{}")
-            x_params.AddValue("model_part_name",settings["model_part_name"])
-            x_params.AddValue("mesh_id",settings["mesh_id"])
-            (x_params.AddEmptyValue("constrained")).SetBool(settings["constrained"][0].GetBool())
-            x_params.AddValue("interval",settings["interval"])
-            x_params.AddValue("value",settings["value"][0])
-            x_params.AddEmptyValue("variable_name").SetString(settings["variable_name"].GetString() + "_X")
-            x_params.AddValue("local_axes",settings["local_axes"])
-            self.aux_processes.append( assign_scalar_variable_process.AssignScalarVariableProcess(Model, x_params) )
-
-        #component Y
-        if(not settings["value"][1].IsNull()):
-            y_params = KratosMultiphysics.Parameters("{}")
-            y_params.AddValue("model_part_name",settings["model_part_name"])
-            y_params.AddValue("mesh_id",settings["mesh_id"])
-            y_params.AddEmptyValue("constrained").SetBool(settings["constrained"][1].GetBool())
-            y_params.AddValue("interval",settings["interval"])
-            y_params.AddValue("value",settings["value"][1])
-            y_params.AddEmptyValue("variable_name").SetString(settings["variable_name"].GetString() + "_Y")
-            y_params.AddValue("local_axes",settings["local_axes"])
-            self.aux_processes.append( assign_scalar_variable_process.AssignScalarVariableProcess(Model, y_params) )
-
-        #component Z
-        if(not settings["value"][2].IsNull()):
-            z_params = KratosMultiphysics.Parameters("{}")
-            z_params.AddValue("model_part_name",settings["model_part_name"])
-            z_params.AddValue("mesh_id",settings["mesh_id"])
-            z_params.AddEmptyValue("constrained").SetBool(settings["constrained"][2].GetBool())
-            z_params.AddValue("interval",settings["interval"])
-            z_params.AddValue("value",settings["value"][2])
-            z_params.AddEmptyValue("variable_name").SetString(settings["variable_name"].GetString() + "_Z")
-            z_params.AddValue("local_axes",settings["local_axes"])
-            self.aux_processes.append( assign_scalar_variable_process.AssignScalarVariableProcess(Model, z_params) )
+        # loop over components X, Y and Z
+        for variable_i in enumerate(["_X", "_Y", "_Z"]):
```

Comment:
last question, why do you need enumerate?
doesn't `for variable_i in ["_X", "_Y", "_Z"]:` also work?

Reply:
I think I understand now, but it is not straight forward, why don't you use sth like
`for variable_i, variable_extension in enumerate(["_X", "_Y", "_Z"])`, 
then you don't have to do the nested list manipulation

Reply:
Just to reduce the size, write less
On my point of view, it is equal